cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java

```diff
@@ -63,9 +66,21 @@ public void execute(Project project) {
         PackageCompilation packageCompilation = project.currentPackage().getCompilation();
         JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(packageCompilation, JdkVersion.JAVA_11);
         DiagnosticResult diagnosticResult = jBallerinaBackend.diagnosticResult();
-        diagnosticResult.diagnostics().forEach(d -> err.println(d.toString()));
+        diagnosticResult.diagnostics().forEach(d -> err.println(convertDiagnosticToString(d)));
         if (diagnosticResult.hasErrors()) {
             throw createLauncherException("compilation contains errors");
         }
     }
+
+    private String convertDiagnosticToString(Diagnostic diagnostic) {
+        LineRange lineRange = diagnostic.location().lineRange();
+
+        LineRange oneBasedLineRange = LineRange.from(
+                lineRange.filePath(),
+                LinePosition.from(lineRange.startLine().line() + 1, lineRange.startLine().offset() + 1),
+                LinePosition.from(lineRange.endLine().line() + 1, lineRange.endLine().offset() + 1));
+
+        return diagnostic.diagnosticInfo().severity().toString() + " [" +
+                oneBasedLineRange.filePath() + ":" + oneBasedLineRange + "] " + diagnostic.message();
```

Comment:
Is this the best way to handle this? :thinking: My concern here is that the `toString()` of the Diagnostic impl is duplicated here. How about creating a copy of the whole Diagnostic instance? Maybe add a builder API to `Diagnostic` to make it easier to create copies of Diagnostic instances?

Reply:
We can go with this for now though, and maybe revisit later.

Reply:
IMO, creating a new diagnostic kinda going out of its definition, isn't it? Here, We only need to change the position and print right ?

Reply:
Yeah, converting the diagnostic to a 1-based position. My concern is this feels like a rather ad hoc solution and duplicates the `toString()` of the diagnostic implementation. My point was whether we should consider having a proper util/API for converting a given diagnostic to 1-based positions. 

But I guess we can go with this for now and revisit this later.

Reply:
Cool